docs/architecture/adr-002-lite-client.md

```diff
@@ -0,0 +1,474 @@
+- how Tendermint data-structures implement the core light client traits
+- how a light node requests information from full nodes and detects conflicts
+
+Note that the architecture of IBC is out of scope, suffice it to say that the
+core libraries (ie. the verification library and the implementation of core
+traits by the Tendermint types) will be re-usable by IBC.
+
+## Decision
+
+Below is a schematic of the components of a light node:
+
+![Light Client Diagram](assets/adr-002-image.png)
+
+We take up the components of the diagram in term.
+
+### State
+
+The light node state contains the following:
+
+- current height (H) - height for the next header we want to verify
+- last header (H-1) - height for the last header we verified
+- current validators (H) - validators for the height we want to verify (including all validator pubkeys and voting powers)
+
+It also includes some configuration, which contains:
+
+- trusting period
+- initial list of full nodes
+- method (sequential or skipping)
+- trust level (if method==skipping)
+
+The node is initialized with a trusted header for some height H-1
+(call this header[H-1]), and a validator set for height H (call this vals[H]).
+
+The node may be initialized by the user with only a height and header hash, and
+proceed to request the full header and validator set from a full node. This
+reduces the initialization burden on the user, and simplifies passing this
+information into the process, but for the state to be properly initialized it
+will need to get the correct header and validator set before starting the light
+client syncing protocol.
+
+The configuration contains an initial list of full nodes (peers).
+For the sake of simplicity, one of the peers is selected as the "primary", while the
+rest are considered "backups". Most of the data is downloaded from the primary,
+and double checked against the backups.
+
+The state is considered "expired" if the difference between the current time and
+the time from the trusted header is greater than a configurable "trusting
+period". If at any point the state is expired, the node should log an error and
+exit - it's needs to be manually reset.
+
+
+### Manager 
+
+For lack of a better name. The Manager co-ordiantes the syncing and is the
+highest level component. We consider two approaches to syncing the light node: sequential and skipping.
+
+#### Sequential Sync
+
+Inital state: 
+
+    - time T
+    - height H 
+    - header[H-1]
+    - vals[H]
+
+Here we describe the happy path:
+
+1) Request header[H], commit[H], and vals[H+1] from the primary, and check that they are well formed and from the correct height
+2) Pass header[H], commit[H], vals[H], and vals[H+1] to the verification library, which
+
+  will:
+    - check that vals[H] and vals[H+1] are correctly reflected in the header
+    - check that the commit is for the header
+    - check that +2/3 of the validators correctly signed the header hash
```

Comment:
Everywhere else you specify which header we are talking about (by explicitly listing a height `H` via `header[H]`). I think it would be good be consistent in this listing, too. (same for validators)